src/riak_kv_quanta.erl

```diff
@@ -0,0 +1,250 @@
+%% @doc This module serves to generate time quanta on multi - (year, month, day, hour, minute,
+%% second) boundaries. The quanta are based on an origin time of Jan 1, 1970 00:00:00 (Unix Epoch).
+%% The function <em>quanta/3</em> takes a time in milliseconds to bucketize, a size of the quanta, and the
+%% units of said quanta. For instance, the following call would create buckets for timestamps on 15
+%% minute boundaries: <em>quanta(Time, 15, m)</em>. The quanta time is returned in milliseconds since the
+%% Unix epoch.
+
+-module(riak_kv_quanta).
+
+-export([quanta/3,
+         timestamp_to_ms/1,
+         ms_to_timestamp/1]).
+
+%% @doc A timestamp in millisconds representing number of millisconds from Unix epoch
+-type time_ms() :: non_neg_integer().
+
+%% @doc The units of quantization available to quanta/3
+-type time_unit() :: y | mo | d | h | m | s.
+
+-type err() :: {error, term()}.
+
+%% @doc The Number of Days from Jan 1, 0 to Jan 1, 1970
```

Comment:
Probably don't need a `@doc` annotation here, it might produce unexpected effects. I'd just remove it and put the comments after/below the define.